src/main/scala/Core/CSR.scala

```diff
@@ -35,14 +35,16 @@ class CSR(implicit val conf: FlexpretConfiguration) extends Module {
     val epc = Input(UInt(32.W)) // PC of uncommitted instruction
     val cause = Input(UInt(CAUSE_WI.W))
     val evecs = Output(Vec(conf.threads, UInt(32.W)))
+    val mepcs = Output(Vec(conf.threads, UInt(32.W)))
     // timing
-    val sleep = Input(Bool()) // valid DU inst
+    val sleep_du = Input(Bool())
+    val sleep_wu = Input(Bool())
     val ie = Input(Bool()) // valid IE inst
     val ee = Input(Bool()) // valid EE inst
     val expire = Output(Bool()) // DU, WU time expired
     val dec_tid = Input(UInt(conf.threadBits.W))
     // privileged
-    val sret = Input(Bool()) // valid sret instruction (ignores exception)
+    val xret = Input(UInt(2.W))
```

Comment:
X here refers to either of M, U, S. I used X instead of M to make it simpler to understand how this can be expanded in the future, but it might just be easier to name it `mret` instead.

Reply:
Why width of 2 and not a boolean as it used to be?

Reply:
I tried to encode the three possible privilege levels RISC-V defines: `mret`, `sret`, and `uret`, and that requires two bits. We are only using `mret` however. I think we discussed this a while ago, whether it was good idea to encode the `sret` and `uret` as well when not using them. And I think we concluded that the two others should be removed from the code. So I'll change the code to remove the two not in use.